src/MongoDB/ReadPreference.c

```diff
@@ -194,12 +193,12 @@ static PHP_METHOD(ReadPreference, __construct)
 
 	/* Separate the tagSets zval, since we may end up modifying it in
 	 * php_phongo_read_preference_prep_tagsets() below. */
-	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling);
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z|a/!a!", &mode, &tagSets, &options) == FAILURE) {
-		zend_restore_error_handling(&error_handling);
-		return;
-	}
-	zend_restore_error_handling(&error_handling);
+	PHONGO_PARSE_PARAMETERS_START(1, 3)
+	Z_PARAM_ZVAL(mode)
+	Z_PARAM_OPTIONAL
+	Z_PARAM_ARRAY_EX(tagSets, 1, 1)
```

Comment:
Noted this applies `separate` for the parsed zval. I believe the Manager and ReadPreference constructors were the only place where this was necessary.